templates/react-native/src/client.ts.twig

```diff
@@ -0,0 +1,410 @@
+                            })
+                        }
+                        break;
+                    case 'error':
+                        throw message.data;
+                    default:
+                        break;
+                }
+            } catch (e) {
+                console.error(e);
+            }
+        },
+        cleanUp: channels => {
+            this.realtime.channels.forEach(channel => {
+                if (channels.includes(channel)) {
+                    let found = Array.from(this.realtime.subscriptions).some(([_key, subscription] )=> {
+                        return subscription.channels.includes(channel);
+                    })
+
+                    if (!found) {
+                        this.realtime.channels.delete(channel);
+                    }
+                }
+            })
+        }
+    }
+
+    /**
+     * Subscribes to {{spec.title | caseUcfirst}} events and passes you the payload in realtime.
+     * 
+     * @param {string|string[]} channels 
+     * Channel to subscribe - pass a single channel as a string or multiple with an array of strings.
+     * 
+     * Possible channels are:
+     * - account
+     * - collections
+     * - collections.[ID]
+     * - collections.[ID].documents
+     * - documents
+     * - documents.[ID]
+     * - files
+     * - files.[ID]
+     * - executions
+     * - executions.[ID]
+     * - functions.[ID]
+     * - teams
+     * - teams.[ID]
+     * - memberships
+     * - memberships.[ID]
+     * @param {(payload: RealtimeMessage) => void} callback Is called on every realtime update.
+     * @returns {() => void} Unsubscribes from events.
+     */
+    subscribe<T extends unknown>(channels: string | string[], callback: (payload: RealtimeResponseEvent<T>) => void): () => void {
+        let channelArray = typeof channels === 'string' ? [channels] : channels;
+        channelArray.forEach(channel => this.realtime.channels.add(channel));
+
+        const counter = this.realtime.subscriptionsCounter++;
+        this.realtime.subscriptions.set(counter, {
+            channels: channelArray,
+            callback
+        });
+
+        this.realtime.connect();
+
+        return () => {
+            this.realtime.subscriptions.delete(counter);
+            this.realtime.cleanUp(channelArray);
+            this.realtime.connect();
+        }
+    }
+
+    async call(method: string, url: URL, headers: Headers = {}, params: Payload = {}): Promise<any> {
+        method = method.toUpperCase();
+
+
+        headers = Object.assign({}, this.headers, headers);
+        headers.Origin = `{{ spec.title | caseLower }}-${Platform.OS}://${this.config.platform}`
+
+        let options: RequestInit = {
+            method,
+            headers,
+            credentials: 'include'
+        };
+
+        if (typeof window !== 'undefined' && window.localStorage) {
+            headers['X-Fallback-Cookies'] = window.localStorage.getItem('cookieFallback') ?? '';
+        }
```

Comment:
I don't think localStorage is going to work:

<img width="1063" alt="image" src="https://github.com/appwrite/sdk-generator/assets/1477010/f01ac920-5b02-480c-beaf-d5b403a201b5">

Reply:
removed as this is not required. Cookies don't suffer from cross-domain issues and we react native handles cookie automatically